### PR TITLE
[#2354] Implement providers soft delete

### PR DESCRIPTION
**Pull request by danielkryska**
- [x] hide a provider with status `deleted` on /providers page 
- [x] remove the provider with status `deleted` from searching
- [x] the provider with status `deleted` can't be removed/updated/edited
- [x] URL in resource description providing to the provider with `deleted` status should be disabled
- [x] A new provider created in Backoffice has the status `published`
- [x] Filter out the providers with status deleted/draft from /services sidebar (Providers multi-select)
- [x] Add status field validation (require)
- [x] Update JMS and PcCreateOrUpdate to use Providers statuses
- [x] When exists service with a status different than delete and is resource organization of the provider, the provider can't be removed 
- [x] secure all subpages od /provider and /backoffice/provider endpoints
- [x] filter out soft-deleted providers from back-office forms
- [x] filter out soft-deleted providers from a list in resource views (`Provided by:`)
- [ ] ~~Add errored status on creating invalid Provider~~

closes: #2354, #2147

### CODE REVIEW

File: app/models/provider.rb [hunk 1]
```diff
@@ -79,6 +86,7 @@ def search_data
     unless legal_entity
       self.legal_status = nil
     end
+    self.status ||= ENV["PROVIDER_IMPORT_STATUS"] || "published"
```
[the remark this answers is not in the document]
Comment (danielkryska) [reply app/models/provider.rb:89]: status = ENV["RESOURCE_IMPORT_STATUS"] || "published"

in `app/services/importers/service.rb` line 77

**Review by kmarszalek — APPROVED** [review]
good job :)

**Review by kmarszalek — APPROVED** [review]

**Comment by Marcelinna** [comment]
Steps to reproduce. 

1. Go to backoffice
2. Delete Provider "AUTHENIX"
3. Go to /providers/authenix - User see message You are not authorized to see this page.
4. Go to /providers/authenix/details  - user see details

I think that the policy of showing providers should be equal to the policy of showing resources. 
Expected result: 404 page is displayed

**Comment by Marcelinna** [comment]
Steps to reproduce:
1. Create a provider eg. "new provider"
2. Add created the previous provider as Provider in a resource form
image
3. Deleted provider "new provider"

Actual result: Provider "new provider" is visible on resource presentation page
Expected result: Provider Resource isn't a required field, shouldn't be displayed 
image
image

**Comment by Marcelinna** [comment]
Steps to reproduce

1. Create provider "new provider"
2. Deleted provider "new provider"
image
3. Go to resource creation form
Actual result: deleted provider can be added as Resource Organisation and Provider
image
Expected result: deleted provider cant be added as Resource Organistation and Provider

File: app/services/provider/destroy.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  def call
+    active_organisation_service = Service.where(resource_organisation_id: @provider.id).where.not(status: :deleted)
+    if active_organisation_service.present?
+      false
+    else
+      @provider.status = :deleted
+      @provider.save(validate: false)
+    end
```
Comment (kmarszalek) [inline app/services/provider/destroy.rb:15]: refresh ES?

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Steps to reproduce

1. Go to Backoffice
2. Create prvider
3. Delete provider
4. See error provider is deleted but user see information - This Provider has resources connected to it, therefore is not possible to remove it.

**Comment by Marcelinna** [comment]
beta:
1. Create a provider in PC
2. Delete provider in Backoffice - provider is deleted.
3. Update provider in PC
4. Provider is again visible in MP

pr:
1. Create a provider in PC
2. Delete provider in Backoffice - provider is deleted
3. Update provider in PC
4. Provider isn't visible in MP

**Review by kmarszalek — APPROVED** [review]

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**